Libplanet/Blockchain/BlockChain.cs

```diff
@@ -518,6 +570,16 @@ DateTimeOffset currentTime
             }
         }
 
+        public void AddTipHandler(EventHandler<long> handler)
+        {
+            _tipChanged += handler;
+        }
+
+        public void RemoveTipHandler(EventHandler<long> handler)
+        {
+            _tipChanged -= handler;
```

Comment:
How about exposing `_tipChanged` as a public property?  Microsoft's WinForms tend to have such APIs.  Read also the official docs' [*Standard .NET event patterns*](https://docs.microsoft.com/en-us/dotnet/csharp/event-pattern).

Reply:
I've considered the way you mentioned, but since `BlockChain<T>.MineBlock()` subscribes `_tipChanged` to seed whether tip is changed, It might work not as expected if user modifies `_tipChanged`.

Reply:
Ah, I was wrong.  I meant a *public event*, not a *public property*, e.g.:

```csharp
public event EventHandler<long> TipChanged;
```